### ISSUE TITLE
discovery: true and authKey

### ISSUE THREAD

**Issue by williamstein** [neumino/rethinkdbdash#133]
When I set both discovery: true and the authKey in the options, I get a bunch of errors about wrong authKey.  It seems that maybe setting discovery:true causes the driver to try to read conf information before properly connecting using the authKey.    Using exactly the same options, but without discovery:true, and everything works fine.  

Any ideas?

**Comment by neumino** [comment]
What are the errors? Stackstrace?

**Comment by williamstein** [comment]
It's not a stack trace.  The error just says that the auth key is incorrect.

**Comment by neumino** [comment]
Just tried with 4 servers, it works fine now.

How many servers do you have? How do you run them? Typically are you running them inside containers? If yes do you start them with `--bind ...`?

**Comment by williamstein** [comment]
I have tried with 1 server and with 3 servers, using both RethinkDB 2.0.4 and 2.1 and in all cases I get the same behavior (see log below).    I use the `bind=all` option in my `/etc/rethinkdb/instances.d/default.conf` (that's the only option except for join for some nodes).   

If I remove discovery:true, but keep the other (hosts) option the same, then everything works fine.   

I'm running the VM's as Google Compute Engine instances running Ubuntu 15.04 (very clean minimal install). 
 
```
Creating a pool connected to db0:28015
Creating a pool connected to db1:28015
Creating a pool connected to db2:28015
Creating a pool connected to 10.240.63.216:28015
Creating a pool connected to 10.240.83.243:28015
Creating a pool connected to 10.240.1.224:28015
Removing pool connected to: db0:28015
Draining the pool connected to db0:28015
Removing pool connected to: db2:28015
Draining the pool connected to db2:28015
Entering slow growth mode
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
Fail to create a new connection for the connection pool. Error:{"message":"Server dropped connection with message: 'ERROR: Incorrect authorization key.\n'."}
...
```

The line `Creating a pool connected to 10.240.63.216:28015` above looks suspicious since why is it trying to connect again to a node that is already there?  

Here's what one of the entries of the server_status table looks like (it does have size 3):
```
{"id":"680f661a-445e-4213-9eff-88e9db2b830e","name":"db1","network":{"canonical_addresses":[{"host":"10.240.63.216","port":29015},{"host":"127.0.0.1","port":29015}],"cluster_port":29015,"hostname":"db1","http_admin_port":8080,"reql_port":28015,"time_connected":"2015-08-16T22:22:14.360Z"},"process":{"argv":["/usr/bin/rethinkdb","--daemon","--config-file","/etc/rethinkdb/instances.d/default.conf","--runuser","rethinkdb","--rungroup","rethinkdb","--pid-file","/var/run/rethinkdb/default/pid_file","-directory","/var/lib/rethinkdb/default/data"],"cache_size_mb":1264.615234375,"pid":2196,"time_started":"2015-08-16T22:15:22.703Z","version":"rethinkdb 2.1.1+1+g989a97~0vivid (GCC 4.9.2)"}}
```

Any hints or ideas?

**Comment by neumino** [comment]
Are 10.240.63.216, 10.240.83.243 and 10.240.1.224 the ips of db0, db1 and db2?

Do here what happens is the driver connects to `db0`, `db1`, and `db2`.
It then retrieve the ips of the servers from the table `server_status`.
To match a connection to a server, we check a match for <ip>:<port> but not for <hostname>:<ip>
So because we don't find a match, we kill the pool connected to `db*` and just use the pool connected to `10.240.*.*`.

I opened https://github.com/neumino/rethinkdbdash/issues/134 to use the hostname to match a pool with a server too.
My guess right now is that one of the `10.240.*.*` ip host a RethinkDB instance with another auth key.

**Comment by neumino** [comment]
I just pushed 2.1.6 with the fix for #134. You should stay connected to db0, db1, db2 and probably won't get the error anymore.

If you still have an issue let me know. My best guess is still that one of the `ips` to which rethinkdbdash connects, hosts another instance of RethinkDB with a different key.
I tried to set up a cluster with an authkey and connect via a hostname that RethinkDB doesn't know to trigger the same behavior and I have no issue reconnecting to the cluster.

**Comment by neumino** [comment]
Closing, if you still have the issue with 2.1.6, feel free to re-open
